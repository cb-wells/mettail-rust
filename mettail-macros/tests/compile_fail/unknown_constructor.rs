use mettail_macros::theory;

theory! {
    name: Invalid,
    
    exports {
        Proc
        Name
    }
    
    terms {
        PZero . Proc ::= "0" ;
        NQuote . Name ::= "@" (Proc) ;
    }
    
    equations {
        // Error: unknown constructor 'Unknown'
        (Unknown) == (PZero)
    }
}

fn main() {}

