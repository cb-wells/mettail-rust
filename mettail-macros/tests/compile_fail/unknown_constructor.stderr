error: expected identifier
  --> tests/compile_fail/unknown_constructor.rs:13:31
   |
13 |         NQuote . Name ::= "@" (Proc) ;
   |                               ^
